### ISSUE TITLE
Cannot reproduce reported SDR & retrain the speaker embedding

### ISSUE THREAD

**Issue by nnbtam** [mindslab-ai/voicefilter#30]
Hello, I have two questions about the implementation.

1. I cannot reproduce the results reported in the README.
I have trained for around > 400k steps on Librispeech 360h + 100h clean dataset, using the embedder provided in this repo.
However, I can only obtain up to a maximum SDR of 5.5.

To obtain data from the Librispeech 360h + 100h, I generate the mixed audios for 360h and 100h separately, then add them together in another folder. Is this the right way when I want to use more data to train the voice filter module?

2. I got worse results when retraining the speaker embedding
I retrained the embedder using the following repo: [Speaker verification](https://github.com/HarryVolek/PyTorch_Speaker_Verification) on 3 datasets: Librispeech, VoxCeleb1, VoxCeleb2.

Theoretically, I expect the voice filter module will benefit from the embedder trained on more data, but the results got even worse. Can you share how you train this embedder?

Thank you in advance!

**Comment by Andrew-Infinite** [comment]
I guess I had an identical result with you, I am at around 4.8 after 76k steps using Librispeech 360h + 100h using the embedding provided.

My current guess is that the mixing generated is random causing the difference in our result. Or, maybe he activated VAD for the data generation.

<img width="1246" height="439" alt="Image" src="https://github.com/user-attachments/assets/b59dc398-2afd-4e77-b577-160d0f974d91" />

**Comment by ChEnYoNgB** [comment]
> 我猜我的结果和你一样，使用 Librispeech 360 小时 + 100 小时提供的嵌入，经过 76k 步后，我的数值在 4.8 左右。
> 
> 我目前的猜测是，生成的混合数据是随机的，这导致了我们结果的差异。或者，他可能启用了 VAD 来生成数据。
> 
> <img alt="图像" width="1246" height="439" src="https://private-user-images.githubusercontent.com/81764717/511571390-b59dc398-2afd-4e77-b577-160d0f974d91.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM5NzIzNjksIm5iZiI6MTc2Mzk3MjA2OSwicGF0aCI6Ii84MTc2NDcxNy81MTE1NzEzOTAtYjU5ZGMzOTgtMmFmZC00ZTc3LWI1NzctMTYwZDBmOTc0ZDkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI0VDA4MTQyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiMjYzN2E1ZmZkMmE3N2YzOGI3NDQyYTlhNTM2MmNlNGViYzBkMWEyZGI4MzA4ZjA0ZGExNzJjMTllOWJmM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CoSs47ZbXrAcbsbfjEaVo9tjyTsPRXMBHZf0lsxyWcc">

Hello, have you made any progress? I have been training for 30,0000 steps but still can't reach the SDR mentioned in the project. If according to your speculation, using randomly mixed data, how can we reproduce the results? Additionally, later I will try using VAD.

**Comment by Andrew-Infinite** [comment]
Sadly, I couldn't. Also, I am training a slightly different model (no bi-lstm), because my main goal were online voicefilter instead.

I had also tried VAD, I didn't quantitatively measure, but I think it actually hurt real world data performance. Especially the case where the targeted voice is not present. 

The best result I obtain so far were averagely 5.6, but again, I am already training a different model.

**Comment by Andrew-Infinite** [comment]
First of all props to the author of the repo, thanks to his public repo I was able to get into Audio research slightly easier.

I think the cause of differences in SDR not reproducible were mainly because of the validation code. I cannot say for sure he didn't test it out separately, but if he was using the validation in the code for SDR, only a single audio from the validation set were use. So if you were lucky to use one that was easy for the model, you can get good SDR really fast, but if it was a difficult one, the SDR wouldn't be able to grow as fast.

If I was correct, take the result as a pinch of salt and try figure out your way to get SDR 10.0  (I am still trying to by the way /.\)
